clarity/src/vm/contexts.rs

```diff
@@ -276,11 +282,11 @@ impl AssetMap {
         asset: &AssetIdentifier,
         amount: u128,
     ) -> Result<u128> {
-        let current_amount = match self.token_map.get(principal) {
-            Some(principal_map) => *principal_map.get(asset).unwrap_or(&0),
-            None => 0,
-        };
-
+        let current_amount = self
+            .token_map
+            .get(principal)
+            .map(|x| x.get(asset).unwrap_or(&0))
+            .unwrap_or(&0);
```

Comment:
This avoids assigning a value twice:

```suggestion
            .map(|x| x.get(asset))
            .flatten()
            .unwrap_or(&0);
```